injectproxy/rules.go

```diff
@@ -208,7 +208,7 @@ func (r *routes) filterRules(lvalue string, resp *apiResponse) (interface{}, err
 		var rules []rule
 		for _, rule := range rg.Rules {
 			for _, lbl := range rule.Labels() {
-				if lbl.Name == r.label && lbl.Value == lvalue {
+				if lbl.Name == r.label && contains(lvalues, lbl.Value) {
```

Comment:
It could use https://pkg.go.dev/sort#SearchStrings
```suggestion
				if lbl.Name == r.label && sort.SearchStrings(lvalues, lbl.Value) {
```

Reply:
> SearchStrings searches for x in a sorted slice of strings and returns the index as specified by Search. The return value is the index to insert x if x is not present (it could be len(a)). 

If we want to find whether a string is in the slice, it must test `lvalues[i] == lbl.Value` separately.

It is still needed to define a function to deal with it. The function will look like something like this:

```golang
func contains(someValues []string, value string) bool {
    i := sort.SearchStrings(someValues, value)
    
    return i < len(someValues) && someValues[i]==value
}
```

Reply:
I have change the code like this:
https://github.com/prometheus-community/prom-label-proxy/blob/95871f0751cae73bde76a8315ac1fc2b7a81a9d1/injectproxy/rules.go#L246-L250

Please tell me if I think wrong.